aip/general/0203.md

```diff
@@ -112,41 +121,42 @@ are:
 - Derived or structured information based on original user input.
 - Properties of a resource assigned by the service which can not be altered.
 
-### Input only
-
-The use of `INPUT_ONLY` indicates that the field is provided in requests and
-that the corresponding field will not be included in output.
-
-Additionally, a field **should** only be described as input only if it is a
-field in a resource message or a field of a message included within a resource
-message. Notably, fields in request messages (a message which only ever acts as
-an argument to an RPC, with a name usually ending in `Request`) **should not**
-be described as input only because this is already implied.
-
-Potential use cases for input only fields (this is not an exhaustive list) are:
+### Required
 
-- The `ttl` field as described in AIP-214.
+The use of `REQUIRED` indicates that the field **must** be present (and set to
+a non-empty value) on the request or resource.
 
-**Warning:** Input only fields are rare and should be considered carefully
-before use.
+A field **should** only be described as required if _either_:
 
-### Immutable
+- It is a field on a resource that a user provides somewhere as input. In this
+  case, the resource is only valid if a "truthy" value is _stored_.
+  - When [creating][aip-133] the resource, a value **must** be provided for the
+    field on the create request.
+  - When [updating][aip-134] the resource, the user **may** omit the field
+    provided that the field is also absent from the field mask, indicating no
+    change to the field (otherwise it **must** be provided).
+- It is a field on a request message (a message that is an argument to an RPC,
+  with a name usually ending in `Request`). In this case, a value **must** be
+  provided as part of the request, and failure to do so **must** cause an error
+  (usually `INVALID_ARGUMENT`).
 
-The use of `IMMUTABLE` indicates that a field on a resource cannot be changed
-after it's creation. This can apply to either fields that are input or outputs,
-required or optional.
+We define the term "truthy" above as follows:
 
-When a service receives an immutable field in an update request (or similar),
-even if included in the update mask, the service **should** ignore the field if
-the value matches, but **should** error with `INVALID_ARGUMENT` if a change is
-requested.
+- For primitives, values other than `0`, `0.0`, empty string/bytes, and `false`
+- For repeated fields maps, values with at least one entry
+- For messages, any message with at least one "truthy" field.
 
-Potential use cases for immutable fields (this is not an exhaustive list) are:
+Fields **should not** be described as required in order to signify:
 
-- Names or IDs which are set on creation and then used as a primary key.
```

Comment:
same comment here, shouldn't we call out that name / id fields should not be marked as required, since the IDENTIFIER annotation will express that?

Reply:
fixed PTAL.